here-naksha-app-service/src/test/java/com/here/naksha/app/service/CreateFeatureTest.java

```diff
@@ -367,4 +379,74 @@ void tc0309_testCreateFeaturesWithUuid() throws Exception {
             .hasStatus(200)
             .hasJsonBody(thirdResponse);
   }
+
+  // TODO : This test is disabled for now due to failure in FibMap for generating payload bigger than 6MB
+  /*
+  @Test
+  @EnabledIf("runBigPayloadTests")
+  void tc0310_testBigRequestPayload() throws Exception {
+    // Test API : POST /hub/spaces/{spaceId}/features
+    // Given: Input base Feature JSON
+    final String basePayload = loadFileOrFail("CreateFeatures/TC0310_createBigFeature/feature_base_payload.json");
+    final XyzFeature feature = JsonUtil.parseJson(basePayload, XyzFeature.class);
+
+    // Given: Big Feature generated from base Feature
+    final long targetBodySize = (5 * 1024 * 1024) - 1024; // keep buffer of 1KB
+    FeatureUtil.generateBigFeature(feature, targetBodySize-feature.serialize().length());
+
+    // Given: Big request payload ready to test
+    final String requestPayload = JsonUtil.toJson(new FeatureCollectionRequest().withFeatures(List.of(feature)));
+    String streamId = UUID.randomUUID().toString();
+
+    assertTrue(requestPayload.length() >= targetBodySize, "Request payload size isn't big enough");
+
+    // When: Big Feature request is submitted to NakshaHub
+    HttpResponse<String> response = getNakshaClient().post("hub/spaces/" + SPACE_ID + "/features", requestPayload, streamId);
+
+    // Then: Perform assertions that we get success response
+    assertThat(response)
+            .hasStatus(200)
+            .hasStreamIdHeader(streamId)
+            .hasJsonBody(requestPayload, "Create Feature response body doesn't match")
+            .hasInsertedCountMatchingWithFeaturesInRequest(requestPayload)
+            .hasInsertedIdsMatchingFeatureIds(null)
+            .hasUuids();
+  }
+  */
+
+  @Test
+  @EnabledIf("runBigPayloadTests")
+  void tc0311_testFixedSize22MBRequestPayload() throws Exception {
+    // Test API : POST /hub/spaces/{spaceId}/features
+    // Given: Big Input request payload of 22MB
+    final long expBodySize = 22 * 1024 * 1024;
+    final Duration timeout = Duration.ofSeconds(30); // bigger timeout for this request
+    final String bodyJson = loadFileOrFail("CreateFeatures/TC0311_create22MBFeature/big_admin_payload_20485579.json");
```

Comment:
I know it's a trade off between CPU and disk storage, but I'm curious if we intend to use FeatureUtil for the 5MB payload test above, should we also use it here? Say to fill 11MB out of 22MB, for example

Reply:
dynamic population of payload randomly faces FibMap specific bug. so, it is not consistent enough to be used as part of Unit Test. And hence, this test was created which directly uses the input payload and doesn't generate it dynamically.

Reply:
ah so it's where the bug surfaces, now I get it. Since it's only for testing, how about some other randomizers, either just native java or other libs? https://www.baeldung.com/java-random-string

Reply:
Issue is not related (as it seems) to generation of random fields. but it is when we add randomly generated data into XyzFeature properties object (which is a Map).